test/parallel/test-http-client-unescaped-path.js

```diff
@@ -1,9 +1,14 @@
 'use strict';
-var common = require('../common');
-var assert = require('assert');
-var http = require('http');
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
 
-assert.throws(function() {
-  // Path with spaces in it should throw.
-  http.get({ path: 'bad path' }, common.fail);
-}, /contains unescaped characters/);
+function* bad() {
+  for (let i = 0; i <= 32; i += 1)
+    yield 'bad' + String.fromCharCode(i) + 'path';
```

Comment:
I initially had that but I felt it was less readable than concatenation.  Consider:
```js
yield 'bad' + String.fromCharCode(i) + 'path';
```
Vs.
```js
yield `bad${String.fromCharCode(i)}path`;
```